### PR TITLE
[Enhancement](stmt-forward) better error msg for follower fe

### LINKED ISSUE #16300: [Enhancement] better error msg for follower fe
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

发往 fellower fe 的请求转发给 master fe 后，如果超时，那么返回客户端的报错信息 和 fe 的日志都会很奇怪，需要改进。

返回用户的报错：
image

follower fe 日志：
image

预期：
无论 client 连接的是 follower fe 上还是 master fe ，都应该返回真正的错误原因，比如“ backend RPC timeout“

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Hastyshell**
# Proposed changes

Issue Number: close #16300

## Problem summary

The error log msg for the FE follower's forward to master failure is ambiguous root exception cause, wether to sql client or in fe log, so we should clarify it. 

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/qe/MasterOpExecutor.java [hunk 1]
```diff
@@ -127,19 +129,28 @@ private void forward() throws Exception {
...
-                throw e;
+                throw exception;
             } else {
                 LOG.warn("Forward statement " + ctx.getStmtId() + " to Master " + thriftAddress + " twice", e);
-                result = client.forward(params);
-                isReturnToPool = true;
+                try {
+                    client.forward(params);
+                    isReturnToPool = true;
+                } catch (TException ex) {
```
Comment (Yukang-Lian) [inline fe/fe-core/src/main/java/org/apache/doris/qe/MasterOpExecutor.java:151]: maybe use `e` so that the format is consistent with all exception?

Comment (TangSiyang2001) [reply fe/fe-core/src/main/java/org/apache/doris/qe/MasterOpExecutor.java:151]: That may result in a naming confict haha.

**Review by Yukang-Lian — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
Hi @TangSiyang2001 , Could you please use text instead of image to show the issue?
And is there any show case after your modification?

**Comment by freemandealer** [comment]
Thanks for your contribution. As a gentle reminder, please show us the error msg output after applying this fix.

**Comment by freemandealer** [comment]
> Thanks for your contribution. As a gentle reminder, please show us the error msg output after applying this fix.

Maybe you can try this:

1) Start two fe processes, and then `show frontends` to see which one is is the follower

2) set insert into timeout short enough to let the backend fail at timeout

3) Send an insert into SQL to the follower fe. When it times out, you should see the error

**Comment by TangSiyang2001** [comment]
> Hi @TangSiyang2001 , Could you please use text instead of image to show the issue? And is there any show case after your modification?

Sure, I'll modify the descriptions and get the show case.

**Comment by TangSiyang2001** [comment]
> > Thanks for your contribution. As a gentle reminder, please show us the error msg output after applying this fix.
> 
> Maybe you can try this:
> 
>     1. Start two fe processes, and then `show frontends` to see which one is is the follower
> 
>     2. set insert into timeout short enough to let the backend fail at timeout
> 
>     3. Send an insert into SQL to the follower fe. When it times out, you should see the error

Thank you for the guidance, I'll work in this way.

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by TangSiyang2001** [comment]
sql client log
```
mysql> insert into lineorder select * from lineorder limit 10000000;
ERROR 1105 (HY000): ForwardToMasterException, msg: org.apache.doris.qe.MasterOpExecutor$ForwardToMasterException: Forward statement 21 to Master TNetworkAddress(hostname:127.0.0.1, port:9020) failed, cause: Unknown exception
```
fe log
```
2023-02-27 01:02:15,733 WARN (mysql-nio-pool-6|140) [StmtExecutor.execute():630] execute Exception. stmt[22, 9bed44b1950b4351-b003c1e13820c44e]
org.apache.doris.qe.MasterOpExecutor$ForwardToMasterException: Forward statement 22 to Master TNetworkAddress(hostname:127.0.0.1, port:9020) failed, cause: Unknown exception
        at org.apache.doris.qe.MasterOpExecutor.forward(MasterOpExecutor.java:137) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.MasterOpExecutor.execute(MasterOpExecutor.java:73) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.forwardToMaster(StmtExecutor.java:695) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.execute(StmtExecutor.java:497) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.execute(StmtExecutor.java:425) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:415) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.ConnectProcessor.dispatch(ConnectProcessor.java:557) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.ConnectProcessor.processOnce(ConnectProcessor.java:794) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.mysql.ReadListener.lambda$handleEvent$0(ReadListener.java:52) ~[doris-fe.jar:1.2-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_352]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_352]
        at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_352]
```
However, since the socket timeout exception will be classified as type: `unkonwn exception` in thrift, rather than `conn timeout` in reality, the result will not be `conn timeout` still. But If we use  `conn timeout` msg to cover the `unkonwn exception`, that may hide the msg of other `unkonwn exception`s, should I enhance it further @freemandealer ?

**Pull request merged**